cardano_node_tests/tests/test_cli.py

```diff
@@ -393,6 +395,29 @@ def _check_stake_snapshot(  # noqa: C901
                 stake_snapshot = cluster_obj.g_query.get_stake_snapshot(
                     stake_pool_ids=expected_pool_ids
                 )
+
+                # Check stake distribution on dbsync
+                # The stake distribution is extracted from the "set" snapshot of the ledger
+                if configuration.HAS_DBSYNC:
+                    current_epoch = cluster_obj.g_query.get_tip()["epoch"]
+
+                    pool_epoch_stake_amount = 0
+                    stake = None
+                    for stake in dbsync_queries.query_epoch_stake(
+                        pool_id_bech32=pool_ids[0], epoch_number=current_epoch
+                    ):
+                        pool_epoch_stake_amount += stake.amount
+
+                    assert (
+                        stake
+                    ), f"No epoch stake distribution found in dbsync relate to pool {pool_ids[0]}"
+
+                    if pool_epoch_stake_amount:
+                        assert (
+                            stake_snapshot["pools"][stake.hash.hex()]["stakeSet"]
+                            == pool_epoch_stake_amount
+                        ), "The epoch stake distribution in dbsync doesn't match stake snapshot"
+
```

Comment:
How about just:
```python
pool_stake_snapshots = next(iter(stake_snapshot["pools"].values()))

db_set_sum = sum(r.amount for r in dbsync_queries.query_epoch_stake(...))
snapshot_set_sum = pool_stake_snapshots["stakeSet"] or 0

assert  db_set_sum == snapshot_set_sum, ...
```

and you can do the same for `current_epoch - 1` and check `stakeMark` as well.

Also let's move this to the end of the test, like
```python
if option == "single_pool" and configuration.HAS_DBSYNC:
    ...
```
and add potential errors to `errors` instead of using `assert`.